src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/CostFunctionsForRootOperators.java

```diff
@@ -0,0 +1,20 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+
+public interface CostFunctionsForRootOperators {
+
+    int determineNumberOfRequests( final PhysicalPlan pp ) throws QueryOptimizationException;
```

Comment:
You're right. It is more logical to take 'PhysicalOperator' as input for these methods. 
However, only passing 'PhysicalOperator' to these methods does not work because the cost of a root operator depends on the 'intermediate result size' of the subquery. 

When determining the cost of a root operator, we need to call **getSubPlan**() by passing 'PhysicalPlan' to these methods. Does it make sense?

Reply:
Ah, yes! You are right.